File: 2025/d3/p1.py
with open("input.txt") as f:
    nbs = []
    for line in f.readlines():
        nbs.append(int(line.strip()))

res = 0
for nb in nbs:
    nb = str(nb)
    max1, index1, max2 = max(nb[:-1]), nb.index(max(nb[:-1])), max(nb[nb.index(max(nb[:-1])) + 1 :])
    res += int(max1 + max2)

print(res)

File: 2025/d3/p1.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <bits/stdc++.h>
using namespace std;

int main(){
    ifstream file("input.txt");

    int res = 0;
    string s;

    while (getline(file, s)) {

        s.erase(remove_if(s.begin(), s.end(), ::isspace), s.end());

        string left = s.substr(0, s.size() - 1);
        char max1 = *max_element(left.begin(), left.end());

        size_t i = left.find(max1);

        char max2 = *max_element(s.begin() + i + 1, s.end());
        res += (max1 - '0') * 10 + (max2 - '0'); // passer de ascii a int
    }
    
    cout << res << '\n';
    return 0;
}